subsys/net/openthread/rpc/client/ot_rpc_commissioning_client.c

```diff
@@ -0,0 +1,145 @@
+		memcpy(aResult.mNetworkName.m8, result.nwk_name, OT_NETWORK_NAME_MAX_SIZE + 1);
+		memcpy(aResult.mExtendedPanId.m8, result.ext_pan_id, OT_EXT_PAN_ID_SIZE);
+		memcpy(aResult.mSteeringData.m8, result.steering_data, OT_STEERING_DATA_MAX_LENGTH);
+		aResult.mPanId = result.pan_id;
+		aResult.mJoinerUdpPort = result.joiner_udp_port;
+		aResult.mChannel = result.channel;
+		aResult.mRssi = result.rssi;
+		aResult.mLqi = result.lqi;
+		aResult.mVersion = result.version;
+		aResult.mIsNative = result.is_native;
+		aResult.mDiscover = result.discover;
+		aResult.mIsJoinable = result.is_joinable;
+	}
+	context = (void *)nrf_rpc_decode_uint(ctx);
+	callback_slot = (otHandleActiveScanResult)nrf_rpc_decode_callback_call(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_decoding_error(OT_RPC_CMD_THREAD_DISCOVER_CB, handler_data);
+		return;
+	}
+
+	if (callback_slot) {
+		callback_slot(pkt_data ? &aResult : NULL, context);
+	}
+
+	nrf_rpc_rsp_send_void(group);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_thread_discover_cb, OT_RPC_CMD_THREAD_DISCOVER_CB,
+			 ot_rpc_thread_discover_cb_rpc_handler, NULL);
+
+otError otThreadDiscover(otInstance *aInstance, uint32_t aScanChannels, uint16_t aPanId,
+			 bool aJoiner, bool aEnableEui64Filtering,
+			 otHandleActiveScanResult aCallback, void *aCallbackContext)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 0;
+	otError error;
+
+	ARG_UNUSED(aInstance);
+
+	cbor_buffer_size += sizeof(uint32_t) + 1;  /* aScanChannels */
+	cbor_buffer_size += sizeof(uint32_t) + 1;  /* aPanId */
+	cbor_buffer_size += 1;                     /* aJoiner */
+	cbor_buffer_size += 1;                     /* aEnableEui64Filtering */
+	cbor_buffer_size += sizeof(uintptr_t) + 1; /* aCallback */
+	cbor_buffer_size += sizeof(uint32_t) + 1;  /* aCallbackContext */
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_encode_uint(&ctx, aScanChannels);
+	nrf_rpc_encode_uint(&ctx, aPanId);
+	nrf_rpc_encode_bool(&ctx, aJoiner);
+	nrf_rpc_encode_bool(&ctx, aEnableEui64Filtering);
+	nrf_rpc_encode_callback(&ctx, aCallback);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)aCallbackContext);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_THREAD_DISCOVER, &ctx, ot_rpc_decode_error,
+				&error);
+
+	return error;
+}
+
+bool otDatasetIsCommissioned(otInstance *aInstance)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	bool result;
+
+	ARG_UNUSED(aInstance);
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, 0);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_DATASET_IS_COMMISSIONED, &ctx,
+				nrf_rpc_rsp_decode_bool, &result);
+
+	return result;
+}
+
+otError otDatasetSetActiveTlvs(otInstance *aInstance, const otOperationalDatasetTlvs *aDataset)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size;
+	otError error;
+
+	ARG_UNUSED(aInstance);
+
+	if (aDataset == NULL || aDataset->mLength > OT_OPERATIONAL_DATASET_MAX_LENGTH) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+	cbor_buffer_size = aDataset->mLength + 1;
```

Comment:
This may be insufficient, you need an extra byte to encode the length itself.
```suggestion
	cbor_buffer_size = aDataset->mLength + 2;
```